### PR TITLE
Destinations: handle collisions between destination states

### PR DESCRIPTION
**Pull request by edgao**
@gisripa this should solve the dest-postgres dup state key thing, though... probably not worth merging until after the cdk stabilizes

also wraps destination-state-related code in try-catch, since we treat state as a best-effort thing.

### CODE REVIEW

**Comment by gisripa** [comment]
Sweet. I'll put it in the queue after this https://github.com/airbytehq/airbyte/pull/36620

**Comment by edgao** [comment]
lmk if you want me to wrap all the state stuff in try-catch return emptyMap(), I didn't add that to this pr but it's probably an easy change

File: airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcDestinationHandler.kt [hunk 1]
```diff
@@ -265,27 +262,32 @@ abstract class JdbcDestinationHandler<DestinationState>(
...
+                    // Sort by updated_at, so that if there are duplicate state,
+                    // the most recent state is the one that gets used.
+                    // That shouldn't typically happen, but let's be defensive.
+                    val updatedAt = it.get(DESTINATION_STATE_TABLE_COLUMN_UPDATED_AT)?.asText()
+                    if (updatedAt != null) {
+                        OffsetDateTime.parse(updatedAt)
+                    } else {
+                        OffsetDateTime.MIN
+                    }
+                }.associate {
```
Comment (jbfbell) [inline airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcDestinationHandler.kt:278]: neat!

**Review by jbfbell — APPROVED** [review]

**Pull request merged**